checkov/terraform/checks/resource/kubernetes/AllowPrivilegeEscalation.py

```diff
@@ -24,6 +25,14 @@ def scan_resource_conf(self, conf) -> CheckResult:
             return CheckResult.UNKNOWN
         spec = spec_list[0]
         if spec:
+            evaluated_keys_path = "spec"
+
+            if spec.get("template") and isinstance(spec.get("template"), list):
+                template = spec.get("template")[0]
+                if template.get("spec") and isinstance(template.get("spec"), list):
+                    spec = template.get("spec")[0]
+                evaluated_keys_path = f'{evaluated_keys_path}/[0]/template/[0]/spec'
```

Comment:
```suggestion
            template = spec.get("template")
            if template and isinstance(template, list):
                template_spec = template[0].get("spec")
                if template_spec and isinstance(template_spec, list):
                    spec = template_spec[0]
                evaluated_keys_path = f'{evaluated_keys_path}/[0]/template/[0]/spec'
```
this way you don't access the `dict` multiple times to retrieve the same value. Same goes for the other checks you updated.

Reply:
Changed this also in the other PRs-